From cd2605cf3e36d6c19334aae78f84a699c40831c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Sun, 10 May 2020 00:08:51 +0300
Subject: [PATCH 3/4] encavcodec: Set the framerate field in addition to
 time_base

The AVCodecContext struct nowadays has got a frame rate field, which
is supposed to be used for indicating the frame rate, allowing the
time_base to be different, for properly describing VFR content.
(There's potentially still some encoders that still interpret
1/time_base as the frame rate though.)
---
 libhb/encavcodec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libhb/encavcodec.c b/libhb/encavcodec.c
index e6899412f..5bef84dbd 100644
--- a/libhb/encavcodec.c
+++ b/libhb/encavcodec.c
@@ -250,6 +250,7 @@ int encavcodecInit( hb_work_object_t * w, hb_job_t * job )
 
     context->time_base.den = fps.num;
     context->time_base.num = fps.den;
+    context->framerate     = fps;
     context->gop_size  = ((double)job->orig_vrate.num / job->orig_vrate.den +
                                   0.5) * 10;
     if ((job->vcodec == HB_VCODEC_FFMPEG_VCE_H264) || (job->vcodec == HB_VCODEC_FFMPEG_VCE_H265))
-- 
2.17.1

